### PR TITLE
ioquake3: fix binaries installation location; bump version

### LINKED ISSUE #69162: Why are ioquake3's executables not placed in bin/
**Describe the bug**
See title.

**To Reproduce**
1. `nix-build -A ioquake3`
2. result/bin/ioquake3*

**Expected behavior**
ioquake3 runs/

### PR DESCRIPTION
**Pull request by drperceptron**
###### Description of changes

Fixes #69162. Bumps version too while we're here. Separate commits.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/games/quake3/ioquake/default.nix [hunk 1]
```diff
@@ -56,6 +57,12 @@ stdenv.mkDerivation {
     mkdir -p $out/baseq3
   '';
 
+  postInstall = ''
+    mkdir $out/bin
+    makeWrapper $out/ioquake3.x86_64 $out/bin/ioquake3
+    makeWrapper $out/ioq3ded.x86_64 $out/bin/ioq3ded
```
Comment (wegank) [inline pkgs/games/quake3/ioquake/default.nix:63]: Should be `.arm64` on aarch64-linux.

Comment (drperceptron) [reply pkgs/games/quake3/ioquake/default.nix]: Thank you. Fixed.

When I realized supported platforms are all `lib.platforms.linux` it became obvious this was something to defer to the package's installer. We now simply specify `COPYBINDIR` to `installFlags`.

**Comment by drperceptron** [comment]
Adding `FULLBINEXT=` to `installFlags` removes the annoying architecture extensions on the binary file names (i.e. `ioquake3` instead of `ioquake3.x86_64`). However the package's documentation consistently references the binaries formatted with an architecture suffix (e.g. `ioquake3.x86_64.exe`) so we should follow their convention unless someone has a strong opinion to the contrary.

**Comment by nagy** [comment]
> unless someone has a strong opinion to the contrary

Provide a symlink in `postInstall` to support both?

**Comment by drperceptron** [comment]
> Provide a symlink in `postInstall` to support both?

My preference is to stick with what ioquake3 documents -- i.e. keep the architecture suffix.
 * [Players Guide](https://ioquake3.org/help/players-guide/)
 * [Sys Admin Guide](https://ioquake3.org/help/sys-admin-guide/)
It's ugly but it's documented. ;-) It's also what we do already.

That said, [Debian](https://packages.debian.org/bullseye/amd64/ioquake3/filelist), [Ubuntu](https://packages.ubuntu.com/kinetic/amd64/ioquake3/filelist), and [Arch](https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=ioquake3-git) all install binaries without an architecture suffix.

I'd rather take a position rather than do both with symbolic links. Having - or not having - the architecture suffix on binaries is an option already supported by the package's installer. Doing both involves unaesthetic code like `ln -s $out/bin/ioquake3.* $out/bin/ioquake3`. Not great. Not terrible.

**Review by trofi — APPROVED** [review]

**Comment by davidak** [comment]
Result of `nixpkgs-review pr 209008` run on i686-linux [1](https://github.com/Mic92/nixpkgs-review)

**Review by davidak — APPROVED** [review]
Change looks good

Builds, runs

**Comment by davidak** [comment]
Result of `nixpkgs-review pr 209008` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li>quake3demo</li>
  </ul>
</details>
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ioquake3</li>
  </ul>
</details>

**Comment by trofi** [comment]
Looks like it broke `quake3wrapper`-based installs like `quake3demo`. Do you want to give it a try and see if it needs a fix?

Reproducer:

```
$ NIXPKGS_ALLOW_UNFREE=1 nix build -f. quake3demo -L
...
quake3-demo> Builder called die: Cannot wrap '/nix/store/khlq2wa0i7rab4vkzvk4pl54lyi6c36d-quake3-demo-1.11-6-ioquake3-unstable-2022-11-24/bin/quake3' because it is not an executable file
```

**Comment by trofi** [comment]
Another problem is that now `quake3` does not start when ran from `PATH`:

```
$ export PATH=$(nix-build -A ioquake3)/bin
ioquake3.x86_64
...
"Failed loading ./renderer_opengl2_x86_64.so: ./renderer_opengl2_x86_64.so: cannot open shared object file: No such file or directory"
```

It normally relies on the wrapper to inject correct paths: `--add-flags "+set fs_basepath ${env}`. I think it's now broken.

**Comment by trofi** [comment]
Proposed wrapper fix as https://github.com/NixOS/nixpkgs/pull/218064

**Pull request merged**